packages/babel-plugin/src/utils/__tests__/Cache.test.tsx

```diff
@@ -0,0 +1,118 @@
+import { Cache, CacheOptions } from '../Cache';
+
+jest.mock('@compiled/utils', () => {
+  return { ...jest.requireActual('@compiled/utils'), hash: () => 'hash-test' };
+});
+
+describe('#Cache', () => {
+  beforeAll(() => {
+    Cache.getUniqueKey = (key: string, namespace?: string) =>
+      namespace ? `${namespace}----${key}` : key;
+  });
+
+  const setup = (options: CacheOptions = {}) => {
+    const cache = new Cache(options);
+
+    return { cache };
+  };
+
+  it('should cache the values with default "cache" option as true', () => {
+    const { cache } = setup();
+
+    const data = { namespace: 'namespace', cacheKey: 'cacheKey', value: () => 10 };
+    const value = cache.load(data);
+
+    expect(cache.getSize()).toBe(1);
+    expect(value).toBe(10);
+  });
+
+  it('should cache the values when "cache" option is true', () => {
+    const { cache } = setup({ cache: true });
+
+    const data = { namespace: 'namespace', cacheKey: 'cacheKey', value: () => 10 };
+    const value = cache.load(data);
+
+    expect(cache.getSize()).toBe(1);
+    expect(value).toBe(10);
+  });
+
+  it('should not cache the values when "cache" option is false', () => {
+    const { cache } = setup({ cache: false });
+
+    const data = { namespace: 'namespace', cacheKey: 'cacheKey', value: () => 10 };
+    const value = cache.load(data);
+
+    expect(cache.getSize()).toBe(0);
+    expect(value).toBe(10);
+  });
+
+  it('should return original cached value when accessed with original key', () => {
+    const { cache } = setup();
+
+    const data = { namespace: 'namespace', cacheKey: 'cacheKey', value: () => 10 };
+
+    cache.load(data);
```

Comment:
why do we have three calls here? can we add one test that asserts that the lazy `value` only gets called once? :)

Reply:
I mean, I was testing the behaviour instead of implementation detail. So if something is cached, it will be returned from cache. Probably I can pass different value with same key and check if value is same as initial? 🤔

Reply:
fair but we haven't tested the behaviour that caching is actually working right? we could set value to a jest mock and assert how many  times its called?